test/e2e/storage/testsuites/volume_group_snapshottable.go

```diff
@@ -172,22 +174,37 @@ func (s *VolumeGroupSnapshottableTestSuite) DefineTests(driver storageframework.
 				init(ctx)
 				createPodsAndVolumes(ctx)
 				ginkgo.DeferCleanup(cleanup)
+				// dc := groupTest.config.Framework.DynamicClient
 
 				snapshot := storageframework.CreateVolumeGroupSnapshotResource(ctx, snapshottableDriver, groupTest.config, pattern, labelValue, groupTest.volumeGroup[0][0].Pvc.GetNamespace(), f.Timeouts, map[string]string{"deletionPolicy": pattern.SnapshotDeletionPolicy.String()})
 				groupTest.snapshots = append(groupTest.snapshots, snapshot)
 				ginkgo.By("verifying the snapshots in the group are ready to use")
-				status := snapshot.Vgs.Object["status"]
+				status := snapshot.VGS.Object["status"]
 				err := framework.Gomega().Expect(status).NotTo(gomega.BeNil())
 				framework.ExpectNoError(err, "failed to get status of group snapshot")
-				volumes := status.(map[string]interface{})["pvcVolumeSnapshotRefList"]
-				err = framework.Gomega().Expect(volumes).NotTo(gomega.BeNil())
+
+				volumeListMap := snapshot.VGSContent.Object["status"].(map[string]interface{})
+				err = framework.Gomega().Expect(volumeListMap).NotTo(gomega.BeNil())
+				framework.ExpectNoError(err, "failed to get volume snapshot list")
+				volumeSnapshotHandlePairList := volumeListMap["volumeSnapshotHandlePairList"].([]interface{})
+				err = framework.Gomega().Expect(volumeSnapshotHandlePairList).NotTo(gomega.BeNil())
 				framework.ExpectNoError(err, "failed to get volume snapshot list")
-				volumeList := volumes.([]interface{})
-				err = framework.Gomega().Expect(len(volumeList)).To(gomega.Equal(groupTest.numVolumes))
+				err = framework.Gomega().Expect(len(volumeSnapshotHandlePairList)).To(gomega.Equal(groupTest.numVolumes))
 				framework.ExpectNoError(err, "failed to get volume snapshot list")
 				claimSize := groupTest.volumeGroup[0][0].Pvc.Spec.Resources.Requests.Storage().String()
-				for _, volume := range volumeList {
+				for _, volume := range volumeSnapshotHandlePairList {
 					// Create a PVC from the snapshot
+					volumeHandle := volume.(map[string]interface{})["volumeHandle"].(string)
+					err = framework.Gomega().Expect(volumeHandle).NotTo(gomega.BeNil())
+					framework.ExpectNoError(err, "failed to get volume handle from volume")
+					uid := snapshot.VGSContent.Object["metadata"].(map[string]interface{})["uid"].(string)
+					err = framework.Gomega().Expect(volumeHandle).NotTo(gomega.BeNil())
+					framework.ExpectNoError(err, "failed to get uuid from content")
+					volumeSnapshotName := fmt.Sprintf("snapshot-%x", sha256.Sum256([]byte(
+						uid+volumeHandle)))
+
+					framework.ExpectNoError(err, "Failed to get volume snapshot from group snapshot")
```

Comment:
This `err` comes from checking UID above, not from getting a snapshot. Reading the rest of the file, I think this line can be removed.